src/utils/offlineRequestHandler.ts

```diff
@@ -217,6 +223,13 @@ export class OfflineRequestHandler {
         if (e instanceof TypeError)
           this.saveRequest(url, options, type, stepKey);
         untrackUnload();
+        // On stale form error, remove beforeunload listener and reload window to
+        // refresh the form version without user data loss warnings.
+        if ((e as any).name == StaleFormError.name) {
```

Comment:
checking the error's name explicitly since `instanceof` will match to `Error` and using the string `StaleFormError` can break this logic if the name is ever changed without also changing it here.